code/game/objects/structures/popout_cake.dm

```diff
@@ -2,6 +2,25 @@
 //A huge cake that can fit a person inside
 //You can cut the cake up to 16 times to receive a cake slice. After that the cake is destroyed and a large cardboard box is left
 
+//LOCKING CATEGORY FOR OBJECTS THAT ARE CLIMBING OUT OF THE CAKE
+/datum/locking_category/popout_cake
+	pixel_y_offset = 1 //First row of pixels on the cake is transparent :^
+
+	var/pixel_y_offset_max = 24
+
+/datum/locking_category/popout_cake/lock(atom/movable/AM)
+	..()
+
+	animate(AM, pixel_y = pixel_y_offset_max, 40) //Increase pixel_y over 4 seconds
```

Comment:
Nope rereading that I think autocorrect did that (I typed that on my phone)

Wew